note: external requirements
  --> $DIR/escape-upvar-nested.rs:21:32
   |
LL |             let mut closure1 = || p = &y;
   |                                ^^^^^^^^^
   |
   = note: defining type: test::{closure#0}::{closure#0} with closure substs [
               i16,
               extern "rust-call" fn(()),
               (&'_#1r i32, &'_#2r mut &'_#3r i32),
           ]
   = note: number of external vids: 4
   = note: where '_#1r: '_#3r

note: external requirements
  --> $DIR/escape-upvar-nested.rs:20:27
   |
LL |           let mut closure = || {
   |  ___________________________^
LL | |             let mut closure1 = || p = &y;
LL | |             closure1();
LL | |         };
   | |_________^
   |
   = note: defining type: test::{closure#0} with closure substs [
               i16,
               extern "rust-call" fn(()),
               (&'_#1r i32, &'_#2r mut &'_#3r i32),
           ]
   = note: number of external vids: 4
   = note: where '_#1r: '_#3r

note: no external requirements
  --> $DIR/escape-upvar-nested.rs:13:1
   |
LL | / fn test() {
LL | |     let x = 44;
LL | |     let mut p = &x;
LL | |
...  |
LL | |     deref(p);
LL | | }
   | |_^
   |
   = note: defining type: test

error[E0597]: `y` does not live long enough
  --> $DIR/escape-upvar-nested.rs:21:40
   |
LL |         let mut closure = || {
   |                           -- value captured here
LL |             let mut closure1 = || p = &y;
   |                                        ^ borrowed value does not live long enough
...
LL |     }
   |     - `y` dropped here while still borrowed
LL | 
LL |     deref(p);
   |           - borrow later used here

error: aborting due to previous error

For more information about this error, try `rustc --explain E0597`.
